RevenueCatUI/Templates/V2/Components/Tabs/TabsComponentView.swift

```diff
@@ -0,0 +1,703 @@
+         tabControlStackViewModels: [StackComponentViewModel]) {
+        self.controlStackViewModel = controlStackViewModel
+        self.tabControlStackViewModels = tabControlStackViewModels
+    }
+
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct TabsComponentView: View {
+
+    @EnvironmentObject
+    private var packageContext: PackageContext
+
+    private let viewModel: TabsComponentViewModel
+    private let onDismiss: () -> Void
+
+    init(viewModel: TabsComponentViewModel, onDismiss: @escaping () -> Void) {
+        self.viewModel = viewModel
+        self.onDismiss = onDismiss
+    }
+
+    var body: some View {
+        LoadedTabsComponentView(
+            viewModel: self.viewModel,
+            parentPackageContext: self.packageContext,
+            onDismiss: self.onDismiss
+        )
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct LoadedTabsComponentView: View {
+
+    @EnvironmentObject
+    private var introOfferEligibilityContext: IntroOfferEligibilityContext
+
+    @EnvironmentObject
+    private var packageContext: PackageContext
+
+    @Environment(\.componentViewState)
+    private var componentViewState
+
+    @Environment(\.screenCondition)
+    private var screenCondition
+
+    private let viewModel: TabsComponentViewModel
+    private let onDismiss: () -> Void
+
+    @StateObject
+    private var tabControlContext: TabControlContext
+
+    @State
+    private var tierPackageContexts: [PackageContext]
+
+    var activeTabViewModel: TabViewModel {
+        return self.viewModel.tabViewModels[self.tabControlContext.selectedIndex]
+    }
+
+    init(viewModel: TabsComponentViewModel,
+         parentPackageContext: PackageContext,
+         onDismiss: @escaping () -> Void) {
+        self.viewModel = viewModel
+        self.onDismiss = onDismiss
+
+        self._tabControlContext = .init(wrappedValue: TabControlContext(
+            controlStackViewModel: viewModel.controlStackViewModel,
+            tabControlStackViewModels: viewModel.tabViewModels.map({ tabViewModel in
+                tabViewModel.stackViewModel
+            })
+        ))
+
+        let variableContext = parentPackageContext.variableContext
+        self._tierPackageContexts = .init(initialValue: viewModel.tabViewModels.map({ tvm in
+            return .init(
+                package: tvm.defaultSelectedPackage,
+                variableContext: .init(
+                    packages: tvm.packages,
+                    showZeroDecimalPlacePrices: variableContext.showZeroDecimalPlacePrices
+                )
+            )
+        }))
+    }
+
+    var body: some View {
+        LoadedTabComponentView(
+            stackViewModel: self.activeTabViewModel.stackViewModel,
+            onChange: { context in
+                self.packageContext.update(
+                    package: context.package,
+                    variableContext: context.variableContext
+                )
+            },
```

Comment:
Each tab will call back to the entire tabs when one of its package contexts is updated (so that the whole paywall knows which new package was selected)